### PR TITLE
add wgpu example with option to make plot the same size as outer window

### PR DESCRIPTION
**Pull request by benmkw**
this is adapted from https://github.com/Yatekii/imgui-wgpu-rs/blob/master/examples/custom_textures.rs and I added the option to make the size of the imgui window stick to the outer window

it's a bit late here so there might be oversights.

happy to apply suggestions as needed (also whether or not this is needed/ wanted here)

### CODE REVIEW

**Comment by 4bb4** [comment]
Thanks, I'll take a look at it once I find a slot, might not be before tomorrow. I've been thinking about where to put the example code for things like `show_configurable_plot` and all that - I'd like to avoid copying it around too much. For example code, I'd otherwise prioritize readability and "standalone-ness" over DRY, but since the API is not super stable yet, I expect I'll have to go back and change all the copies a couple of times.

**Comment by benmkw** [comment]
Sure, so I put your example code in ui.rs (other than that there really was not much to do except maybe getting the context). Maybe this ui.rs could be shared between different examples although I'm not exactly sure how that would work without making it hard to copy an example. But at the very least updating these ui files could be a copy paste thing.

I'll also update the example to imgui-wgpu 0.11 which was just released

**Comment by benmkw** [comment]
i just tried this on windows and compared to the glium example (which also broke for me if text scaling was set on windows), this works with resizing and everything perfectly :)

**Comment by smerkli** [comment]
Yea text scaling has been wonky for me with the glium code (that I had copied out of the imgui repo, I have not even written that). Maybe I should check if that code got improvements in imgui-rs itself in the meantime.

**Comment by 4bb4** [comment]
Yea text scaling has been wonky for me with the glium code (that I had copied out of the imgui repo, I have not even written that). Maybe I should check if that code got improvements in imgui-rs itself in the meantime.

File: implot-wgpu-examples/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "wgpu_plotting"
```
Comment (4bb4) [inline implot-wgpu-examples/Cargo.toml]: Probably this should be `implot-wgpu-examples`?

File: implot-wgpu-examples/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        }
+
+        platform.handle_event(imgui.io_mut(), &window, &event);
+    });
+}
+
+fn main() {
+    let event_loop = EventLoop::new();
+    let window = winit::window::Window::new(&event_loop).unwrap();
+    // Temporarily avoid srgb formats for the swapchain on the web
```
Comment (4bb4) [inline implot-wgpu-examples/src/main.rs]: Are you building this to WASM, or is this just a generic comment from example code?

Comment (benmkw) [reply implot-wgpu-examples/src/main.rs]: I replaced the sample code with the one from https://github.com/Yatekii/imgui-wgpu-rs/blob/master/examples/hello_world.rs with some minor changes which IMO make it more succinct.

**Comment by 4bb4** [comment]
@benmkw I think I have an idea on how to merge this: We could make one examples crate, and then have features that decide which backend to use. We can then write a nice README on how to use the individual backends by enabling separate features. This will allow us to keep just one copy of the backend-unrelated example code, but still keep the example standalone - it'll be a bit "fatter" than individual examples for each backend, but if we make a good enough README, it should be easy to copy the example and delete out the stuff that is not needed.

What do you think?

**Comment by benmkw** [comment]
I thought we could maybe have one crate as a library which exposes render(ui) and then have a crate for each backend in the examples, thus the code is still easy to copy paste for users and there is still only one place to make updates to implot examples. 

The only implot specific code in the backend would be the context creation although that might not even be necessary, I’m on mobile right now :D

**Comment by 4bb4** [comment]
Hmn that could also work, yes. I've already screwed with the examples again in the meantime because I've extended the Y axis selection API today, so anything that keeps the examples code only copied once is fine for me. I'm also looking into what's up with imgui + Rust + WASM at the moment, so there will likely be yet another backend path added in the near future.

**Comment by benmkw** [comment]
cool, thanks for your work :) So I'm not so sure about the feature flag thing because I think it makes it harder to copy code for others thats why I would suggest separate projects for each backend.

fwiw wgpu is currently apparently not yet really suitable for web so maybe wait 6 months and then this may be an option for the web as well :)

**Comment by 4bb4** [comment]
> cool, thanks for your work :) So I'm not so sure about the feature flag thing because I think it makes it harder to copy code for others thats why I would suggest separate projects for each backend.

Let's take your approach then, we create one example crate per implementation and keep the actual rendering code in one crate that we import in the others. Do you want to take a shot at this, or should I merge your code and go about it myself?

> fwiw wgpu is currently apparently not yet really suitable for web so maybe wait 6 months and then this may be an option for the web as well :)

I... did not really get this far yet, I'm still messing with the WASM build of imgui itself - see https://github.com/Gekkio/imgui-rs/issues/342 for more info. But good to know that'll require work as well, so I'm mentally prepared :D

**Comment by benmkw** [comment]
I think its best if you do the reorg of the existing code and examples such that you like it and then I will adapt or complain if it does not work :D

**Comment by 4bb4** [comment]
(Update: this has not been forgotten, I just haven't gotten around to it yet)

**Pull request merged**